File: python_files/raylib/shape_survivors/ui.py
import pyray as p

import utils
from player import Player


def draw_player_xp_bar(player: Player) -> None:
    margin: float = 50.0
    width: float = p.get_screen_width() - (margin * 2)
    height: float = 30.0
    rx: float = margin
    ry: float = 10

    # progress bar
    utils.draw_progress_bar(
        min_value=player.xp.xp,
        max_value=player.xp.xp_next,
        bounds_rect=p.Rectangle(rx, ry, width, height),
        border_thickness=5.0,
        border_color=p.BLACK,
        background_color=p.LIGHTGRAY,
        fill_color=p.SKYBLUE,
    )

    # player level
    p.draw_text(
        f"LV {player.level.current}",
        int(rx + width - 100),
        int(ry + 5),
        20,
        p.BLACK,
    )

File: cpp_files/raylib/shape_survivors/src/ui.cpp
#include "ui.hpp"
#include "utils.hpp"

void ui_player_xp_bar(const Player &player)
{
    const float margin = 50.0f;
    const float width = GetScreenWidth() - (margin * 2);
    const float height = 40.0f;
    const float x = GetScreenWidth() / 2.0f - width / 2.0f;
    const float y = 10.0f;

    const Rectangle bounds = {x, y, width, height};

    draw_progress_bar(player.xp.xp, player.xp.xp_next, bounds, 5.0f, BLACK,
                      LIGHTGRAY, SKYBLUE);

    center_and_draw_text(OriginType::top_left,
                         "LV " + std::to_string(player.level.current),
                         {bounds.x + bounds.width / 2.0f, bounds.y,
                          bounds.width / 2.0f, bounds.height},
                         20.0f, 2.0f, BLACK);
}
